src/main/java/org/distributed/systems/chord/actors/Node.java

```diff
@@ -19,10 +35,31 @@ public Receive createReceive() {
         log.info("Received a message");
 
         return receiveBuilder()
-                .matchEquals("printit", p -> {
-                    log.info("The address of this actor is: " + getSelf()); // Log my reference
-                    getSender().tell("Got Message", getSelf()); // Acknowledge
-                }).build();
+                .match(NodeJoinMessage.class, nodeJoinMessage -> fingerTableService.addSuccessor(nodeJoinMessage.getNode()))
+                .match(PutValueMessage.class, putValueMessage -> {
+                    String key = putValueMessage.getKey();
+                    Object value = putValueMessage.getValue();
+                    log.info("key, value: " + key + " " + value);
+                    valueStore.put(putValueMessage.getKey(), putValueMessage.getValue());
```

Comment:
TODO:: or propogate it if this node is not responsible of this key (this is not necessary fot this PR tho)

Reply:
Use key and value reference  in line 43